src/com/itacademy/eshop/shop/ShoppingCart.java

```diff
@@ -26,4 +26,18 @@ public String getTotalPrice() {
          */
         throw new UnsupportedOperationException("getTotalPrice() is not implemented yet");
     }
+
+    public void addProduct(Product someProduct) {
+        products.add(someProduct);
+    }
+
+    public void removeProductByName(String someProduct) {
```

Comment:
loads of naming comments but just want to help you get ready for clean code. someProduct is fine. but you can use just product as well for the name of the method parameter